data/npc/scripts/bank.lua

```diff
@@ -0,0 +1,347 @@
+		else
+			count[cid] = getMoneyCount(msg)
+			npcHandler:say("So you would like me to change " .. count[cid] * 100 .. " of your gold coins into " .. count[cid] .. " platinum coins?", cid)
+			npcHandler.topic[cid] = 15
+		end
+	elseif npcHandler.topic[cid] == 15 then
+		if msgcontains(msg, "yes") then
+			if player:removeItem(ITEM_GOLD_COIN, count[cid] * 100) then
+				player:addItem(ITEM_PLATINUM_COIN, count[cid])
+				npcHandler:say("Here you are.", cid)
+			else
+				npcHandler:say("Sorry, you do not have enough gold coins.", cid)
+			end
+		else
+			npcHandler:say("Well, can I help you with something else?", cid)
+		end
+		npcHandler.topic[cid] = 0
+	elseif msgcontains(msg, "change platinum") then
+		npcHandler:say("Would you like to change your platinum coins into gold or crystal?", cid)
+		npcHandler.topic[cid] = 16
+	elseif npcHandler.topic[cid] == 16 then
+		if msgcontains(msg, "gold") then
+			npcHandler:say("How many platinum coins would you like to change into gold?", cid)
+			npcHandler.topic[cid] = 17
+		elseif msgcontains(msg, "crystal") then
+			npcHandler:say("How many crystal coins would you like to get?", cid)
+			npcHandler.topic[cid] = 19
+		else
+			npcHandler:say("Well, can I help you with something else?", cid)
+			npcHandler.topic[cid] = 0
+		end
+	elseif npcHandler.topic[cid] == 17 then
+		if getMoneyCount(msg) < 1 then
+			npcHandler:say("Sorry, you do not have enough platinum coins.", cid)
+			npcHandler.topic[cid] = 0
+		else
+			count[cid] = getMoneyCount(msg)
+			npcHandler:say("So you would like me to change " .. count[cid] .. " of your platinum coins into " .. count[cid] * 100 .. " gold coins for you?", cid)
+			npcHandler.topic[cid] = 18
+		end
+	elseif npcHandler.topic[cid] == 18 then
+		if msgcontains(msg, "yes") then
+			if player:getFreeCapacity() >= getMoneyWeight(count[cid]) then
+				if player:removeItem(ITEM_PLATINUM_COIN, count[cid]) then
+					player:addItem(ITEM_GOLD_COIN, count[cid] * 100)
+					npcHandler:say("Here you are.", cid)
+				else
+					npcHandler:say("Sorry, you do not have enough platinum coins.", cid)
+				end
+			else
+				npcHandler:say("Whoah, hold on, you have no room in your inventory to carry all those coins. I don't want you to drop it on the floor, maybe come back with a cart!", cid)
+			end
+		else
+			npcHandler:say("Well, can I help you with something else?", cid)
+		end
+		npcHandler.topic[cid] = 0
+	elseif npcHandler.topic[cid] == 19 then
+		if getMoneyCount(msg) < 1 then
+			npcHandler:say("Sorry, you do not have enough platinum coins.", cid)
+			npcHandler.topic[cid] = 0
+		else
+			count[cid] = getMoneyCount(msg)
+			npcHandler:say("So you would like me to change " .. count[cid] * 100 .. " of your platinum coins into " .. count[cid] .. " crystal coins for you?", cid)
+			npcHandler.topic[cid] = 20
+		end
+	elseif npcHandler.topic[cid] == 20 then
+		if msgcontains(msg, "yes") then
+			if player:getFreeCapacity() >= getMoneyWeight(count[cid]) then
```

Comment:
I would say `if player:getFreeCapacity() >= getMoneyWeight(count[cid]) then` should be used when changing coins from platinum to gold (done) or from crystal to platinum (not done) because the weight is increasing, and not in this case (platinum to crystal)

Reply:
indeed, I added it on the wrong check, corrected it now, thanks!